komodo-relational/src/main/java/org/komodo/relational/teiid/internal/TeiidImpl.java

```diff
@@ -881,121 +879,58 @@ public CachedTeiid importContent(UnitOfWork transaction) throws KException {
         ServerManager mgr = ServerManager.getInstance(getRepository());
         CachedTeiid cachedTeiid = mgr.createCachedTeiid(transaction, this);
 
-        TeiidInstance teiidInstance = getTeiidInstance(transaction);
-        if (teiidInstance == null) {
-            throw new KException(Messages.getString(Messages.Relational.TEIID_INSTANCE_ERROR));
-        }
+        // Gets a teiid instance and connects if not connected
+        TeiidInstance teiidInstance = getConnectedTeiidInstance(transaction);
 
+        // Get the teiid instance artifacts and add them to CachedTeiid
         try {
-            teiidInstance.connect();
-            if (! teiidInstance.isConnected()) {
-                throw new KException(Messages.getString(Messages.Relational.TEIID_INSTANCE_CONNECTION_ERROR));
-            }
-
             Collection<TeiidVdb> instVdbs = teiidInstance.getVdbs();
-            if (instVdbs == null)
-                instVdbs = Collections.emptyList();
+            if (instVdbs == null) instVdbs = Collections.emptyList();
 
             Collection<TeiidDataSource> instDataSrcs = teiidInstance.getDataSources();
-            if (instDataSrcs == null)
-                instDataSrcs = Collections.emptyList();
+            if (instDataSrcs == null) instDataSrcs = Collections.emptyList();
 
             Collection<TeiidTranslator> instTranslators = teiidInstance.getTranslators();
-            if (instTranslators == null)
-                instTranslators = Collections.emptyList();
+            if (instTranslators == null) instTranslators = Collections.emptyList();
 
             Collection<String> instDataSourceTypeNames = teiidInstance.getDataSourceTypeNames();
-            if (instDataSourceTypeNames == null)
-                instDataSourceTypeNames = Collections.emptyList();
+            if (instDataSourceTypeNames == null) instDataSourceTypeNames = Collections.emptyList();
 
-            //
-            // Process the vdbs
-            //
+            // Add VDBs to CachedTeiid
             for (TeiidVdb teiidVdb : instVdbs) {
                 if (teiidVdb == null)
                     continue;
 
                 // Export the vdb content into a string
                 String content = teiidVdb.export();
-                if (content == null)
-                    continue;
-
```

Comment:
True. I did check that already. However, relying on an internal method for such a check seems implicit. The checking for null and continuing here is explicit and more performant since the loop can immediately finish the iteration and move on.